From a9296ecae9e6c2f7389c6d1a72dc762f19765d2c Mon Sep 17 00:00:00 2001
From: Vikash Chandola <vikash.chandola@intel.com>
Date: Thu, 30 Jun 2022 10:28:05 +0000
Subject: [PATCH] Ignore VR sensor readings if content is 0xFFFF

As per NM 6.0 specification SPS FW will return 0xFF in certain
erroneous condition with error code stating transaction successful.
BMC need to ignore such readings.

Signed-off-by: Vikash Chandola <vikash.chandola@intel.com>
Upstream-Status: Pending
---
 src/IpmbSensor.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/IpmbSensor.cpp b/src/IpmbSensor.cpp
index 9908341..f7cdf54 100644
--- a/src/IpmbSensor.cpp
+++ b/src/IpmbSensor.cpp
@@ -332,6 +332,17 @@ void IpmbSensor::checkThresholds(void)
 
 bool IpmbSensor::processReading(const std::vector<uint8_t>& data, double& resp)
 {
+    if (isReadMe && data.size() >= 5)
+    {
+        // Mark VR sensor reading as failure if byte at index 3 and 4
+        // are 0xFF.
+        if (data[3] == 0xFF && data[4] == 0xFF)
+        {
+            std::cerr << name << " value is 0xFFFF. Marking error\n";
+            markFunctional(false);
+            return false;
+        }
+    }
     switch (readingFormat)
     {
         case (ReadingFormat::byte0):
-- 
2.25.1

